rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -126,17 +132,222 @@ void Player::play_messages_from_queue()
   } while (!is_storage_completely_loaded() && rclcpp::ok());
 }
 
+//Dynamic Alignment as Fastrtps
+unsigned long alignment(unsigned long data_size, unsigned long last_data_size, unsigned long current_position)
+{
+  return data_size > last_data_size ? (data_size - current_position % data_size) & (data_size-1):0;
```

Comment:
can we do this vendor specific thing here?

how can we use the dds vendor implementation instead of using copied code in here

Reply:
How is this alignment related to time? Just trying to keep the set of changes semantically separated.

Reply:
in our version of this feature, we are changing the timestamp inside the serialized message directly, that way we achieve a "simulated" time while publishing messages via the player class. 
This is related to issue #299

Reply:
Because the offset in typesupport does not provides actual offset in fastrtps serialized data. (It provides the offset in machine memory, using the alignment of machine) In order to figure out the actual offset of "Header", we must calculate the offset according to Fastrtps alignment rule.